Sources/Regex/ASTTransform.swift

```diff
@@ -14,19 +15,31 @@ extension AST {
       return .groupTransform(group, recurse(component), transform: transform)
     case .quantification(let quantifier, let component):
       return .quantification(quantifier, recurse(component))
-    case .character(let c):
-      return .character(c)
-    case .unicodeScalar(let u):
-      return .unicodeScalar(u)
+    case .atom(.char(let c)):
+      return .atom(.char(c))
+    case .atom(.scalar(let u)):
+      return .atom(.scalar(u))
     case .characterClass(var cc):
       cc.matchLevel = level
       return .characterClass(cc)
-    case .any:
-      return .any
-    case .empty:
-      return .empty
-    case .trivia:
-      return .trivia
+
+    case .any, .trivia, .quote, .empty: return self
```

Comment:
@rxwei would there be any difference for `.any` to return `self` vs a `.any`? I'm not sure how this was switching any-grapheme to any-scalar.

I'm failing inside `testLegacyMatchLevel`, where both the grapheme-semantic AST and the scalar-semantic AST are identical because it's just the `..` regex. . The match level is otherwise not associated with the program AFAICT, and thus the compiled programs are identical between scalar and grapheme-semantic levels.

Reply:
Ok, it's because it's above the below check to convert character classes....